### PR TITLE
Add robot as another parameter to ground truth operators.

### PR DESCRIPTION
**Pull request by ashayathalye**
Adds a robot as a parameter to the ground truth operators for multistep-cover and updates the holding predicate to check both the block and the robot state. After this change, the learned operators through nsrt_learning have the robot as one of their parameters. This is important for option learning. Our learned options need to be able to access the robot state, which can only be done if the learned operators have the robot as a parameter.

### CODE REVIEW

File: src/approaches/oracle_approach.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def get_gt_nsrts(predicates: Set[Predicate],
     elif CFG.env == "cover_typed_options":
         nsrts = _get_cover_gt_nsrts(options_are_typed=True)
     elif CFG.env == "cover_multistep_options":
-        nsrts = _get_cover_gt_nsrts(options_are_typed=True,
+        nsrts = _get_cover_gt_nsrts(options_are_typed=True, more_params=True,
```
Comment (tomsilver) [inline src/approaches/oracle_approach.py:38]: can we make this variable name more descriptive? maybe `include_robot_in_holding`?

**Review by tomsilver — COMMENTED** [review]
lgtm thanks! let's let @ronuchit review too

File: src/approaches/oracle_approach.py (hunk 1 above)
Comment (ashayathalye) [reply src/approaches/oracle_approach.py]: Good point, done.

File: src/envs/cover.py [hunk 2]
```diff
@@ -748,6 +752,13 @@ def _get_hand_regions(self, state: State) -> List[Tuple[float, float]]:
                  state.get(targ, "x")+state.get(targ, "width")/10))
         return hand_regions
 
+
```
Comment (ronuchit) [inline src/envs/cover.py:755]: remove blank line

File: src/envs/cover.py [hunk 3]
```diff
@@ -686,7 +689,8 @@ def _Pick_terminal(self, s: State, m: Dict, o: Sequence[Object],
                        p: Array) -> bool:
         # Pick is done when we're holding the desired object.
         del m, p  # unused
-        return self._Holding_holds(s, o)
+        block = o[0]
```
Comment (ronuchit) [inline src/envs/cover.py:692]: Write this as `block, = o` instead to automatically allow the code to crash if the length of `o` is not what we would expect (i.e., 1).

File: src/envs/cover.py [hunk 4]
```diff
@@ -639,7 +642,7 @@ def _create_initial_state(self, rng: np.random.Generator) -> State:
             # [is_block, is_target, width, x]
             data[target] = np.array([0.0, 1.0, width, x])
         # [x, y, grip]
```
Comment (ronuchit) [inline src/envs/cover.py:644]: update this comment to reflect new field you're adding in the next line

File: src/approaches/oracle_approach.py [hunk 5]
```diff
@@ -131,29 +138,38 @@ def pick_sampler(state: State, rng: np.random.Generator,
             ub = float(state.get(b, "pose") + state.get(b, "width")/2)
             ub = min(ub, 1.0)
         return np.array(rng.uniform(lb, ub, size=(1,)), dtype=np.float32)
+
```
Comment (ronuchit) [inline src/approaches/oracle_approach.py:141]: remove blank line (or, make it consistently be there across the file if you'd like)

File: src/approaches/oracle_approach.py [hunk 6]
```diff
@@ -131,29 +138,38 @@ def pick_sampler(state: State, rng: np.random.Generator,
...
     if options_are_typed:
         option_vars = [target]
         option = Place
     else:
         option_vars = []
         option = PickPlace
-    preconditions = {LiftedAtom(IsBlock, [block]),
-                     LiftedAtom(IsTarget, [target]),
-                     LiftedAtom(Holding, [block])}
+
```
Comment (ronuchit) [inline src/approaches/oracle_approach.py:157]: remove blank line

File: src/approaches/oracle_approach.py [hunk 7]
```diff
@@ -131,29 +138,38 @@ def pick_sampler(state: State, rng: np.random.Generator,
...
     add_effects = {LiftedAtom(HandEmpty, []),
                    LiftedAtom(Covers, [block, target])}
-    delete_effects = {LiftedAtom(Holding, [block])}
+    if include_robot_in_holding:
+        preconditions = {LiftedAtom(IsBlock, [block]),
+                     LiftedAtom(IsTarget, [target]),
```
Comment (ronuchit) [inline src/approaches/oracle_approach.py:162]: fix indentation of this line and the one immediately after

**Review by ronuchit — COMMENTED** [review]
LGTM, just some minor comments (2-3 mins of work)

**Comment by ashayathalye** [comment]
Addressed all comments.

**Pull request merged**